protected/views/dataset/view.php

```diff
@@ -746,3 +748,40 @@ class="btn btn-green btn-block claim-button"
 
     });
 </script>
+<script>
+    function goToPage() {
+        var targetPageNumber = document.getElementById('pageNumber').value;
+        var pageID = <?php echo $model->identifier?>;
+        //To validate page number
+        var userInput = parseInt(targetPageNumber);
+        var max = <?php echo $files->getDataProvider()->getPagination()->getPageCount() ?>;
+        //To output total pages
+        // console.log(max);
+        var min = 1;
+        if (userInput >= min && userInput <= max) {
+            console.log("Valid page number!");
+        }else if (userInput > max) {
+            targetPageNumber = max;
+            console.log("Error, return to " + max);
+        } else if (userInput < min) {
+            targetPageNumber = min;
+            console.log("Error, return to " + min);
+        }
+        // var targetUrlArray = Array.apply(null, Array(5)).map(function(_,i) { return window.location.pathname.split("/")[i]});]
+        // Create array with default values
+        //TODO: This array is hardcoded
```

Comment:
Is there more work to do here?

Reply:
The hardcoded array serves the purpose and works well, no more work is needed  at this stage.